### ISSUE TITLE
Use current versions of Autorest and its extensions to generate code

### ISSUE THREAD

**Issue by stevengum** [Microsoft/botbuilder-dotnet#1207]
## Version
4.0.0-4.2.x

## Describe the bug
Connector and Schema libraries are not generated with the newest/newer versions of autorest. 

For 4.3 we should bump the version of autorest used, and also evaluate whether to make the jump from [@microsoft.azure/autorest.csharp](https://www.npmjs.com/package/@microsoft.azure/autorest.csharp) to [@microsoft.azure/autorest.csharp-v2](https://www.npmjs.com/package/@microsoft.azure/autorest.csharp-v2)

## Additional context
Bug tracking JS side of things: Microsoft/botbuilder-js#665

[bug]

**Comment by stevengum** [comment]
Additional work includes improving the infrastructure around generating the Connector and Schema libraries to avoid relying on any one person's machine.

**Comment by stevengum** [comment]
PR #1249 is targeting master, I'll cut another PR against the 4.2 branch once the PR is merged.
